File: for-local-test/python/169.majority-element.py
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        majority = nums[0]
        count = 0
        for num in nums:
            if count == 0:
                majority = num
            if num == majority:
                count += 1
            else:
                count -= 1
        return majority

if __name__ == "__main__":
    solution = Solution()
    nums = [2,2,1,1,1,2,2]
    print(f"Majority element: {solution.majorityElement(nums)}")  # Output: Majority element: 2

File: for-local-test/169.majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int majority = nums[0];
        int count = 0;
        for (int num : nums){
            if (count == 0) majority = num;
            if(num == majority) count++;
            else count--;
        }
        return majority;
    }

    public static void main(String[] args){
        Solution sol = new Solution();
        int[] nums = {2,2,1,1,1,2,2};
        System.out.println(sol.majorityElement(nums));
    }
}

